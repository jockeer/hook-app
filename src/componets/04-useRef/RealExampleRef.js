
import React, { useState } from 'react';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks'
import '../01-useState/counter.css'

const RealExampleRef = () => {

    const [show, setShow] = useState(false)
    return (  
        <>
        <h1>RealExampleRef</h1>
        <hr />
        {show && <MultipleCustomHooks />}

        <button className='btn btn-primary mt-4' onClick={() => {
            setShow(!show)
        }}>
            Show / Hide
        </button>
        </>
    );
}
 
export default RealExampleRef;