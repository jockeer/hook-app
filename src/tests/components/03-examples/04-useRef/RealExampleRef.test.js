import '@testing-library/jest-dom'
import React from 'react';
import {shallow} from 'enzyme'
import RealExampleRef from '../../../../componets/04-useRef/RealExampleRef'


describe('Prueba RealExampleRef', () => {  

    const wrapper = shallow(<RealExampleRef />)

    test('Debe mostrar el componente correctamente', () => { 
    
        expect(wrapper).toMatchSnapshot()
        
    })

    test('Debe mostrar el componente <MultipleCustomHooks />', () => { 
        wrapper.find('button').simulate('click')
    
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true)
        
    })
    

})