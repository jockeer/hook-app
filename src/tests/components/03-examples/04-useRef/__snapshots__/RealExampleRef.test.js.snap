// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba RealExampleRef Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-4"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</Fragment>
`;
